samples/powerplay/src/main/cpp/PowerPlayMultiPlayer.cpp

```diff
@@ -0,0 +1,102 @@
+
+using namespace oboe;
+using namespace parselib;
+
+PowerPlayMultiPlayer::PowerPlayMultiPlayer() = default;
+
+PowerPlayMultiPlayer::~PowerPlayMultiPlayer() = default;
+
+void PowerPlayMultiPlayer::MyPresentationCallback::onPresentationEnded(oboe::AudioStream *oboeStream) {
+    __android_log_print(ANDROID_LOG_INFO, TAG,
+                        "==== MyPresentationCallback() called with gain %f",
+                        mParent->getGain(0)
+    );
+}
+
+void PowerPlayMultiPlayer::setupAudioStream(int32_t channelCount, oboe::PerformanceMode performanceMode) {
+    __android_log_print(ANDROID_LOG_INFO, TAG, "setupAudioStream()");
+    mChannelCount = channelCount;
+
+    openStream(performanceMode);
+}
+
+bool PowerPlayMultiPlayer::openStream(oboe::PerformanceMode performanceMode) {
+    __android_log_print(ANDROID_LOG_INFO, TAG, "openStream()");
+    mLastPerformanceMode = performanceMode;
+
+    // Use shared_ptr to prevent use of a deleted callback.
+    mDataCallback = std::make_shared<MyDataCallback>(this);
+    mErrorCallback = std::make_shared<MyErrorCallback>(this);
+    mPresentationCallback = std::make_shared<MyPresentationCallback>(this);
+
+    // Create an audio stream
+    AudioStreamBuilder builder;
+    builder.setChannelCount(mChannelCount)
+            ->setDataCallback(mDataCallback)
+            ->setErrorCallback(mErrorCallback)
+            ->setPresentationCallback(mPresentationCallback)
+            ->setFormat(AudioFormat::Float)
+            ->setSampleRate(48000)
+            ->setPerformanceMode(performanceMode)
+            ->setFramesPerDataCallback(128)
+            ->setSharingMode(SharingMode::Exclusive);
+
+    Result result = builder.openStream(mAudioStream);
+    if (result != Result::OK) {
+        __android_log_print(
+                ANDROID_LOG_ERROR,
+                TAG,
+                "openStream failed. Error: %s", convertToText(result));
+        return false;
+    }
+
+    if (!OboeExtensions::isMMapUsed(mAudioStream.get())) {
+        constexpr int32_t kBufferSizeInBursts = 2; // Use 2 bursts as the buffer size (double buffer)
+        result = mAudioStream->setBufferSizeInFrames(mAudioStream->getFramesPerBurst() * kBufferSizeInBursts);
+        if (result != Result::OK) {
+            __android_log_print(
+                    ANDROID_LOG_WARN,
+                    TAG,
+                    "setBufferSizeInFrames failed. Error: %s", convertToText(result)
+            );
+        }
+    }
+
+    mSampleRate = mAudioStream->getSampleRate();
+    return true;
+}
+
+
+void PowerPlayMultiPlayer::triggerUp(int32_t index) {
+    mAudioStream->pause();
+}
+
+void PowerPlayMultiPlayer::triggerDown(int32_t index, oboe::PerformanceMode performanceMode) {
+    if (index < mNumSampleBuffers) {
+        mSampleSources[index]->setPlayMode();
+    }
+    if (performanceMode != mLastPerformanceMode) {
+        teardownAudioStream();
+        openStream(performanceMode);
```

Comment:
It should validate if the opening is successful or not. Otherwise, the calling startStream at line 101 will cause app crash.